services/dashboard/views/model/streamingdag.js

```diff
@@ -0,0 +1,201 @@
+/*
+ * Licensed under the Apache License, Version 2.0
+ * See accompanying LICENSE file.
+ */
+'use strict';
+
+angular.module('dashboard.streamingdag', ['dashboard.metrics'])
+
+  .factory('StreamingDag', ['Metrics', function (Metrics) {
+
+    /** The constructor */
+    function StreamingDag(id, processors, levels, edges) {
+      this.id = id;
+      this.processors = {};
+      this.processorsLevels = {};
+      this.edges = {};
+      this.meter = {};
+      this.histogram = {};
+      this.lastUpdated = null;
+
+      processors.map(function (item) {
+        this.processors[item[0]] = item[1];
+      }, /* thisArg */ this);
+
+      levels.map(function (item) {
+        this.processorsLevels[item[0]] = item[1];
+      }, /* thisArg */ this);
+
+      edges.map(function (item) {
+        var source = item[0];
+        var target = item[2];
+        var type = item[1];
+        var id = source + '_' + target;
+        this.edges[id] = {source: source, target: target, type: type};
+      }, /* thisArg */ this);
+    }
+
+    StreamingDag.prototype = {
```

Comment:
If we do #458 we may want to convert this to scala. Add something like
//TODO Try and convert to scala for #458